src/network.c

```diff
@@ -645,7 +616,8 @@ int write_ssl(SSL* ssl, const char* buf, int amount) {
  * @param conn the Connection
  * @param buf buffer with data to write
  * @param amount the size of the data
- * @return The amount of bytes written to the Connection
+ * @return The amount of bytes written to the Connection.
+ *         -1 when it gets an unrecoverable error, just like write().
  */
 int writen_any(Connection* conn, const void* buf, int amount) {
```

Comment:
I believe the original busy loop bug was caused by a combination of SIGPIPE being ignored and this while loop having no exit condition. Breaking from the while loop still depends on the SIGPIPE handler killing the process (or similar).

Would it be worth while documenting this? Or, adding a guarantee that the loop cannot continue indefinitely (like the 5 tries previously in send_msg_any)?